src/porepy/compositional/materials.py

```diff
@@ -0,0 +1,534 @@
+
+        frozen = ["name", "units", "constants_in_SI", "_initialized"] + list(
+            type(self).SI_units.keys()
+        )
+        if name in frozen and self._initialized:
+            raise FrozenInstanceError(
+                f"Cannot asign to field {name}. Names, units and material parameters"
+                + " are frozen once set."
+            )
+        else:
+            super().__setattr__(name, value)
+
+    def to_units(self: _Constants, units: pp.Units) -> _Constants:
+        """Utility to quickly convert constants to new units.
+
+        Parameters:
+            units: A new unit system. Note that ``units`` must cover all SI units
+                declared in :attr:`SI_units` of the constants (child) class.
+
+        Returns:
+            A new instance of of the data class using this method, with parameters
+            converted according to ``units``.
+
+        """
+        return type(self)(name=self.name, units=units, **self.constants_in_SI)
+
+
+_Constants = TypeVar("_Constants", bound=Constants)
+"""Type variable for Constants-like objects. Mainly used to type the units conversion
+of :class:`Constants` correctly."""
+
+
+@dataclass(kw_only=True, eq=False)
+class FluidComponent(Constants, Component):
+    """Material data class for fluid components.
+
+    It declares parameters relevant for fluid-like components, which are expected of a
+    fluid by the remaining framework, especially flow and transport equations.
+
+    This class is intended for 1 fluid component only. Fluid mixtures with multiple
+    components require multiple sets of constants to define individual fluid components.
+
+    This class is used as the default representation of fluid components inside a
+    mixture.
+
+    """
+
+    SI_units: ClassVar[dict[str, str]] = dict(
+        {
+            "density": "kg * m^-3",
+            "molar_mass": "kg * mol^-1",
+            "critical_pressure": "Pa",
+            "critical_temperature": "K",
+            "critical_specific_volume": "m^3 * kg^-1",
+            "acentric_factor": "-",
+            "pressure": "Pa",
+            "temperature": "K",
+            "compressibility": "Pa^-1",
+            "specific_heat_capacity": "J * kg^-1 * K^-1",
+            "thermal_expansion": "K^-1",
+            "viscosity": "Pa * s",
+            "thermal_conductivity": "W * m^-1 * K^-1",
+            "normal_thermal_conductivity": "W * m^-1 * K^-1",
+        }
+    )
+
+    acentric_factor: number = 0.0
+
+    compressibility: number = 0.0
+
+    critical_pressure: number = 1.0
+
+    critical_specific_volume: number = 1.0
+
+    critical_temperature: number = 1.0
+
+    density: number = 1.0
+
+    molar_mass: number = 1.0
+
+    normal_thermal_conductivity: number = 1.0
+
+    thermal_conductivity: number = 1.0
+
+    thermal_expansion: number = 0.0
+
+    specific_heat_capacity: number = 1.0
+
+    viscosity: number = 1.0
+
+
+@dataclass(kw_only=True)
+class SolidConstants(Constants):
```

Comment:
Should this be SolidComponent for consistency?

Reply:
I would say no, or not yet. The class does not inherit from component and has no fraction.